portal-impl/src/com/liferay/portal/upload/LiferayFileItem.java

```diff
@@ -137,6 +138,24 @@ public int getSizeThreshold() {
 	}
 
 	@Override
+	public File getStoreLocation() {
+		DeferredFileOutputStream dfos = null;
+
+		try {
+			dfos = (DeferredFileOutputStream)getOutputStream();
+		}
+		catch (IOException ioe) {
+			_log.error(ioe, ioe);
+		}
+
+		if (dfos == null) {
+			return null;
+		}
+
+		return dfos.getFile();
+	}
+
```

Comment:
@robertoDiaz 

if `getStoreLocation()` is not overriden, it gets called in Apache Commons fileUload's `DiskFileItem` class.
`getStoreLocation()` in `DiskFileItem` looks like this

    ```
public File getStoreLocation() {
        if (dfos == null) {
            return null;
        }
        if (isInMemory()) {
        	return null;
        }
        return dfos.getFile();
    }

```

There is no code in `DiskFileItem` where `dfos` would be closed.